lib/filterForNonReserved.js

```diff
@@ -0,0 +1,15 @@
+function filterForNonReserved (reserved, options) {
+  // Filter out properties that are not reserved.
+  // Reserved values are passed in at call site.
+
+  var object = Object.create(null)
+  for (var i in options) {
```

Comment:
I wish there is a `if (options.hasOwnProperty(i))` check here too.   The tricky thing about adding that check is it might break existing code using it.